contracts/InterchainTokenService.sol

```diff
@@ -707,7 +707,7 @@ contract InterchainTokenService is
      * @param amount The amount for the call contract with token.
      */
     function _checkPayloadAgainstGatewayData(bytes memory payload, string calldata tokenSymbol, uint256 amount) internal view {
-        (, bytes32 tokenId, , , uint256 amountInPayload) = abi.decode(payload, (uint256, bytes32, uint256, uint256, uint256));
+        (, bytes32 tokenId, , , uint256 amountInPayload) = abi.decode(payload, (uint256, bytes32, bytes, bytes, uint256));
```

Comment:
You should test this but I am pretty sure there is no difference between `uint256` and `bytes` when decoding if you ignore them, but using `bytes` costs more.

Reply:
I was able to confirm using `bytes` costs more gas.
```
(, bytes32 tokenId, , , uint256 amountInPayload) = abi.decode(payload, (uint256, bytes32, bytes, bytes, uint256));

vs

(, bytes32 tokenId, , , uint256 amountInPayload) = abi.decode(payload, (uint256, bytes32, uint256, uint256, uint256));
```

Hence, just added a comment here.